### PR TITLE
Fairlearn Website Re-design

### PR DESCRIPTION
**Pull request by LeJit**
Updates to Fairlearn website landing page. The content layout design is done, but we're waiting for the new content to fill placeholder sections.

### CODE REVIEW

**Comment by romanlutz** [comment]
Awesome progress!

By clicking through the "build-webpage" check you can see the artifacts including index.html, e.g., currently it's https://2421-133444044-gh.circle-artifacts.com/0/docs/_build/html/index.html

A few things come to mind:
- "We fair because we care" - That wasn't intended, was it?
- lorem ipsum all over the place
- Generally we capitalize the first letter and everything else is lowercase in headers unless it's names. Example: Getting Started with Fairlearn --> Getting started with Fairlearn.
- Resources section doesn't link anywhere and text isn't for the parts there. I imagine that's because you wanted to get the landing page right first. However, we can't check this in unless everything works. How are you planning to stage this?
- quickstart link at the top is broken
- Fairlearn logo size is different on landing page vs other pages
- Hovering over the community elements (Gitter, Twitter, etc.) makes the icon disappear (except for StackOverflow somehow) and the elements move
- copyright should probably still be there

**Comment by LeJit** [comment]
Hey Roman,

For the first two points, I'm still waiting for @MiroDudik to finalize the actual content we want, so I put loren ipsum placeholders. He recommended I open a pull request, so others can see the work-in-progress and give feedback.

The "we fair because we care" tagline is a joke placeholder, and a play on Monster Inc's "We scare because we care" slogan.

> Hovering over the community elements (Gitter, Twitter, etc.) makes the icon disappear (except for StackOverflow somehow) and the elements move

This is strange. It works correctly on my local preview (the social media icons turn white to be visible against the blue background). I'll look into this.

> copyright should probably still be there
The design spec I was given had a Microsoft-free copyright. I believe we're also trying to finalize the new governance in accordance with the launch of the new website.

Your other points are minor fixes that I can easily incorporate when I have the time.

**Comment by romanlutz** [comment]
> > copyright should probably still be there
> The design spec I was given had a Microsoft-free copyright. I believe we're also trying to finalize the new governance in accordance with the launch of the new website.

Right. I didn't mean MSFT copyright, but copyright in general.

**Comment by hildeweerts** [comment]
To what detail would you like to have feedback on the current design? I think there's still some tweaking w.r.t. colors/margins etc. that could improve this draft, but not sure whether that's the best use of time in this stage.

I would suggest to replace the icons with either higher resolution png or svg files. At least on my screen they're a bit pixelly right now. 

I have the same issue as Roman with hovering of the elements. When you hover over Github element the icon is missing and the element becomes smaller, causing the next module to move up.

image

> The "we fair because we care" tagline is a joke placeholder, and a play on Monster Inc's "We scare because we care" slogan.
 
Thank you for the giggles :D

**Comment by MiroDudik** [comment]
The initial draft of the content here. Please comment! @romanlutz @hildeweerts @LeJit @nessamilan @mmadaio

File: docs/static_landing_page/index.html [hunk 1]
```diff
@@ -112,221 +90,107 @@ <h6 class="gray">Mitigate</h6></a>
...
+						and how to use Fairlearn's metrics and algorithms
+						while considering the AI system's broader societal context.
+					</p>
   				</div>
+				<!-- Use Case -->
+				<div class="col-6 col-lg-6">
+					<h6 class="text-left pl-3">Use case | Financial services</h6>
+					<h3 class="text-left pl-3 dark">Assessment and mitigation of fairness issues in credit-card default models</h3>
+					<p class="text-left pl-3">
+						Dirth ma, harellan. Ma banal enasalin. Mar solas ena mar din. Need more text to test line wrapping and bar split thingy.
```
Comment (romanlutz) [inline docs/static_landing_page/index.html:117]: This is just a reminder to replace this filler text :-)

**Comment by LeJit** [comment]
I fixed some of the smaller issues to the website (alignment, social media cards, etc): https://2437-133444044-gh.circle-artifacts.com/0/docs/_build/html/index.html

Let me know what you think!

**Comment by nessamilan** [comment]
@LeJit  This is great progress, thank you for all your hard work on this, it looks GREAT!!! 

Below are a few issues to fix, listed in priority order:

1. **Github Button (header) for accessibility**:
     - Update Github to an active button that includes the Github icon
     - For accessibility include the same hover state as other buttons, an underline on text (down the road I'd recommend a button color change :-))
     - Use SVG for icon so it's not blurry
     Screen Shot 2021-03-10 at 2 29 27 PM


2. **Hero headline**: Update line-height from 47px to 88px so text doesn't overlap:

     Screen Shot 2021-03-10 at 2 28 54 PM


3. **Getting started**: Add an input box background as a visual indicator for copy paste. Right now it's reads through with the text above without a clear distinction as an action vs. plain text:
     Screen Shot 2021-03-10 at 2 29 05 PM


4. **Social icons**: Replace with SVG assets so they aren't blurry:
     Screen Shot 2021-03-10 at 2 29 13 PM


5. **Sociotechnical description**: Update paragraph break to a 20px space after:
     Screen Shot 2021-03-10 at 2 28 59 PM


6. **Resources**: Add 20px padding right of icon, or a margin of 20px between icon and headline:
     
     Screen Shot 2021-03-10 at 2 29 23 PM

**Comment by LeJit** [comment]
Latest version of the website: https://2442-133444044-gh.circle-artifacts.com/0/docs/_build/html/index.html

@nessamilan @MiroDudik @romanlutz @hildeweerts

**Comment by MiroDudik** [comment]
Some remaining issues (going top-to-bottom)

1. I think that the hero block would look better if it was left-aligned with the text?
2. We should also make the navbar left-aligned with text?
3. The headings "Fairness is sociotechnical" and "Use case | ... " should be in all caps. Also, I'm going to tweak the description of the use case a bit and make it point to the notebook. [I can take care of these, since these are just simple rewrites.]
4. The vertical spacing in the middle of "Fairness is sociotechnical" is too large. It should look visually the same as vertical spacing between paragraphs in "Get started with Fairlearn"
5. The vertical spacing below the horizontal lines above "Get started with Fairlearn" and above "Resources" is too large.
6. In "Get started with Fairlearn", there needs to be more space above "Fairlearn can be installed". I think that @nessamilan also suggested using a slightly smaller font (to match the Footer font). If it's a pain, you don't need to fiddle with the font, but spacing would help clarity.
7. The code sample needs to have some padding around it (see @nessamilan's proposal).
8. In "Resources", I think that the icons are too loosely connected with themes. I think that for the current version we should just drop the icons, and circle back to it in follow-ups.
9. The shaded backgrounds should go left-to-right without any padding. We have it correct for the hero block, but there's some spurious padding for "Join the community" and "Contribute to Fairlearn."
10. Vertical spacing below "Join the community" seems too small. The titles "Get started with Fairlearn", "Resources", "Join the community", "Contribute to Fairlearn" should be the same font/size and they should have the same amount of vertical whitespace below them and the content text.
11. Vertical spacing between the text "Fairlearn community ..." and rounded square blocks underneath is too large, it should be similar to the vertical spacing between "Join the community" and "Fairlearn community ... ".
12. When the rounded square blocks are highlighted, spurious underline appears. Also, except for the Gitter icons, the regular and white versions of other Icons do not quite match, which is causing "StackOverflow" title and "Twitter" title to move verticaly by a few pixels when the blocks are highlighted.
13. "Contribute to Fairlearn" has small vertical spacing issues: there should be less vertical space above "Contribute to Fairlearn" and there should be more vertical space above the button "Contributor guide". Basically the text "Fairlearn is ... " should be vertically centered between "Contribute to Fairlearn" and "Contributor Guide".
14. Footer: The copyright should be right-justified, it should say "Copyright (c) 2021 Fairlearn contributors." Then the twitter icon should be left-justified (and starting at the same position as text in other blocks, so things are visually left-aligned).

**Comment by LeJit** [comment]
@MiroDudik I think I fixed most of the issues you listed. Let me know if there's anything more I need to change: https://2452-133444044-gh.circle-artifacts.com/0/docs/_build/html/index.html

**Comment by nessamilan** [comment]
## Design bugs for 3/17

### References
[Open Source Design File](https://www.figma.com/file/wt76z0M87RBDDEoyAEpD2g/Fairlearn-Redesign?node-id=1597%3A365)
[Type Ramp](https://www.figma.com/file/wt76z0M87RBDDEoyAEpD2g/?node-id=2025%3A0) 
[Examples: Content + Grid alignment](https://www.figma.com/file/wt76z0M87RBDDEoyAEpD2g/?node-id=1949%3A52398)

## [Bugs](https://www.figma.com/file/wt76z0M87RBDDEoyAEpD2g/?node-id=2229%3A1443)

1. **Buttons**: Inconsistent Padding
Buttons_Inconsistent-padding

2. **Community Cards**: Incorrect H-tag weight and interaction
Community_Type+Interaction

3. **Use case fonts**: Incorrect label + headline style
Type_Incorrect-sizes


4. **H-Tag weights throughout**: Incorrect weight; new type-ramp tag added for the hero module called "Page Title"
H-Tag weights


5. **Grid alignment**: Various content container mis-alignments and incorrect widths
Container_Alignment

**Review by hildeweerts — COMMENTED** [review]
Suggestion: 

"When making a decision to approve or decline a loan, financial services organizations use a variety of models, including a model that predicts the applicant's probability of default. These predictions are sometimes used to automatically reject or accept an application, directly impacting both the applicant and the organization.  

In this scenario, fairness-related harms may arise when the model makes more mistakes for some groups of applicants compared to others. In this use case example, we use Fairlearn to assess how different groups, defined in terms of their sex, are affected and how the observed disparities may be mitigated."

**Comment by MiroDudik** [comment]
> Suggestion:
> 
> "When making a decision to approve or decline a loan, financial services organizations use a variety of models, including a model that predicts the applicant's probability of default. These predictions are sometimes used to automatically reject or accept an application, directly impacting both the applicant and the organization.
> 
> In this scenario, fairness-related harms may arise when the model makes more mistakes for some groups of applicants compared to others. In this use case example, we use Fairlearn to assess how different groups, defined in terms of their sex, are affected and how the observed disparities may be mitigated."

I like @hildeweerts 's suggestion! @LeJit : can you implement when addressing the issues pointed out by @nessamilan?

**Comment by hildeweerts** [comment]
Detail comment: some of the buttons have a right arrow, others don't. Is there a reason for this design choice? I'd be in favor of removing the arrows all together.

**Comment by nessamilan** [comment]
> Detail comment: some of the buttons have a right arrow, others don't. Is there a reason for this design choice? I'd be in favor of removing the arrows all together.

That's a good call out. The arrows were part of a more extended system we'll have on other pages for links + arrows.  I have no problem with buttons being consistent, and without arrows (even with this rationale, @hildeweerts I think it was a mistake on my part not having it on the contribute button!) 

@LeJit I will reconcile the designs to remove the arrow from the buttons, which may also make the spacing issue above much easier to fix!

**Comment by LisaIbanez** [comment]
Use case text looks great! @hildeweerts @MiroDudik  Miro, totally get not being able to pare down further cause of visual balance. Thought about making some of the text more concrete, but I realized the text below the lead banner/headline on the page captures it perfectly. So seems fine to leave text as is, but I am reminded of Hilde's point on Thurs that it seems like a button could be added since there's still noticeable empty space. I wonder if it could just be the get started/quick start button that displays further down on the page--not sure you need the lead up text in the "get started with fairlearn" section especially since it pushes the button/link further down and the button is probably a pretty intuitive call to action. Either way, I noticed that on the Quickstart page reaching out on Gitter if there are any issues isn't mentioned (at least not early on), might be nice to call it out in the same way it's currently called out on the "get started..." section of the homepage.

**Comment by nessamilan** [comment]
Hi all! How are we in terms of homepage content? Do we feel confident in what's there, so I can copy into a draft for our illustrator? @MiroDudik @LeJit @LisaIbanez @hildeweerts

**Comment by MiroDudik** [comment]
@nessamilan : Yes, the current version is the one to give to illustrators!

This is the current version:
https://2486-133444044-gh.circle-artifacts.com/0/docs/_build/html/index.html

I have fixed alignment issues and a couple of other small things. I have talked with @LeJit and he'll need to switch to working on the tutorial for SciPy. There are still various small issues, but unless somebody is willing to take over, I think that we should just merge this (with the understanding that we'll continue iterating on @nessamilan 's full design).

So, this is my proposal:
* Merge this version
* Create an issue "Front page design" that would link to the unresolved issues. As far as I can tell there are two comments that have not been fully resolved:
  * https://github.com/fairlearn/fairlearn/pull/709#issuecomment-801363856
  * https://github.com/fairlearn/fairlearn/pull/709#issuecomment-803449782

@nessamilan , @LisaIbanez , @hildeweerts , @romanlutz : What do you think?

**Review by romanlutz — APPROVED** [review]

**Comment by MiroDudik** [comment]
@hildeweerts : Can you approve? This is the final version, which should work reasonably on both mobile and desktop:
https://2522-133444044-gh.circle-artifacts.com/0/docs/_build/html/index.html

**Review by hildeweerts — APPROVED** [review]

**Pull request merged**